src/Hexagon.java

```diff
@@ -5,15 +5,19 @@
     private TerrainType terrainType;
     private int tileID;
     private int level;
-    private int numVillagersOnTop;
+    private int numberOfVillagersOnTop;
```

Comment:
Perhaps this could be a boolean rather than an integer, since if we know the level and if it has villagers then we know how many villagers there are?  Up to you.